### PR TITLE
feat: Implement new content-utils package

### LINKED ISSUE #82: [new-package] Content collection injection
Double trouble:
- Inject in the file system
- Inject virtually

Thanks to @BryceRussell for the ideas

### PR DESCRIPTION
**Pull request by Fryuni**
- Closes #82

### CODE REVIEW

**Comment by Fryuni** [comment]
If you are wondering why this types have to be so complicated:
https://github.com/Fryuni/inox-tools/blob/f11436e3a72964e281d6299f8be9db0d8625ca7e/packages/content-utils/src/runtime/fancyContent.ts#L5-L40

BLAME @florian-lefebvre!!

**Comment by Fryuni** [comment]
These people that don't export their types...

File: packages/content-utils/src/runtime/injector.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+				`Content collection "${key}" overrides a collection injected by an integration.`,
+				'Try to use a different collection name.'
+			);
+		}
+
+		const originFn = collection[FANCY_COLLECTION_MARKER];
+
+		if (!Object.is(originFn, injectedCollection)) {
+			throw new AstroError(
+				// TODO: Report which integration added the collection.
```
Comment (OliverSpeir) [inline packages/content-utils/src/runtime/injector.ts:24]: Looking forward to seeing this in action

**Review by OliverSpeir — APPROVED** [review]
🚀

File: packages/content-utils/src/runtime/fancyContent.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
+/// <reference types="astro/client" />
+
+import { defineCollection as defineNative, z, type SchemaContext } from 'astro:content';
+
+// Copied from Starlight because Astro core doesn't export their definitions :(
```
Comment (florian-lefebvre) [inline packages/content-utils/src/runtime/fancyContent.ts:5]: Do you think it's worth opening a PR in astro for this?

File: packages/content-utils/src/utility.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+	installPluginOnce.do(() => {
+		addVitePlugin(params, {
+			plugin: plugin(configFile),
+			warnDuplicated: true,
+		});
+	});
+
+	watchDirectory(params, import.meta.dirname);
+
```
Comment (florian-lefebvre) [inline packages/content-utils/src/utility.ts:47]: I think dirname is not available for all node versions, `fileURLToPath(dirname(import.meta.url))` does the same

**Review by florian-lefebvre — COMMENTED** [review]
ngl I mostly understand nothing but manage to leave a couple of comments lol

**Review by BryceRussell — APPROVED** [review]
LGTM! Thanks again for taking the lead on this issue and making a utility everyone can use, I learned a lot! 😄

File: packages/content-utils/src/utility.ts (hunk 3 above)
Comment (Fryuni) [reply packages/content-utils/src/utility.ts:47]: This should have been removed. Thanks!

File: packages/content-utils/src/runtime/fancyContent.ts (hunk 2 above)
Comment (Fryuni) [reply packages/content-utils/src/runtime/fancyContent.ts:5]: Maybe, I'll bring this up on Discord

**Pull request merged**